solved the problem

Avinash11a opened the pull request.

gcnit commented on snake_ladder.cpp:
```diff
@@ -0,0 +1,81 @@
+#include<bits/stdc++.h>
```
Please create multiple classes

gcnit commented on snake_ladder.cpp:
```diff
@@ -0,0 +1,81 @@
+int main()
+{
+    srand(time(0));
+    int snk, lad, per; // Number of snakes, ladders and persons
```
Please use proper variable names

gcnit commented on snake_ladder.cpp:
```diff
@@ -0,0 +1,81 @@
+    int c = 0;
+    while(1)
+    {
+        int x = (rand()%6)+1; // Random dice value generator (1-6)
```
Please make the code modular

gcnit replied:
should have been a method whose job would have to roll a dice and return a number

gcnit commented:
Please follow the submission guidelines mentioned in the document. You need to post the link of the PR as comment on the fb post.

gcnit commented on snake_ladder.cpp:
```diff
@@ -0,0 +1,81 @@
+        {
+            initial = person[c].second;
+            person[c].second = snakes[final];
+            cout << person[c].first << " rolled a " << x << " and moved from " << initial << " to " << person[c].second << endl;
```
why repeat print logic so many times?

gcnit commented on snake_ladder.cpp:
```diff
@@ -0,0 +1,81 @@
+        int initial;
+        int final = person[c].second + x;
+        /*Checking whether at next position a snake exist or not*/
+        if(final<=100 && snakes.find(final)!=snakes.end())
```
why repeat final <= 100 check so many times?

gcnit commented on snake_ladder.cpp:
```diff
@@ -0,0 +1,81 @@
+            cout << person[c].first << " rolled a " << x << " and moved from " << initial << " to " << person[c].second << endl;
+            if(person[c].second==100)
+            {
+                cout << person[c].first << " wins the game" << endl;
```
why repeat this print logic multiple times?

Avinash11a commented:
Thank you for your feedback. I will try to overcome these mistakes in the
next rounds.

On Sat, Sep 7, 2019 at 10:13 PM Gaurav Chandak <notifications@github.com>
wrote:

> *@gcnit* commented on this pull request.
> ------------------------------
>
> In snake_ladder.cpp
> <https://github.com/workattech/mock-machine-coding-1/pull/3#discussion_r321975644>
> :
>
> > +        /*Checking whether at next position a snake exist or not*/
> +        if(final<=100 && snakes.find(final)!=snakes.end())
> +        {
> +            initial = person[c].second;
> +            person[c].second = snakes[final];
> +            cout << person[c].first << " rolled a " << x << " and moved from " << initial << " to " << person[c].second << endl;
> +        }
> +        /*Checking whether at next position a ladder exist or not*/
> +        else if(final<=100 && ladders.find(final)!=ladders.end())
> +        {
> +            initial = person[c].second;
> +            person[c].second = ladders[final];
> +            cout << person[c].first << " rolled a " << x << " and moved from " << initial << " to " << person[c].second << endl;
> +            if(person[c].second==100)
> +            {
> +                cout << person[c].first << " wins the game" << endl;
>
> why repeat this print logic multiple times?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/workattech/mock-machine-coding-1/pull/3?email_source=notifications&email_token=AIS7T3SZ5L7YB53CXRNCSKDQIPKZJA5CNFSM4IUP3ADKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCD7X2GI#pullrequestreview-285179161>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIS7T3T67RICCFKW3UOPRJLQIPKZJANCNFSM4IUP3ADA>
> .
>

gcnit commented:
It'd be great if you can provide us some feedback on the mock machine coding round. Your feedback will help us learn and improve.
https://forms.gle/JiXiEW6ycPUoBcSS7
